js/personalize.admin.js

```diff
@@ -20,35 +20,21 @@
         })
       });
 
-      /*
-      @todo This updates data as dirty when input is clicked on but not changed.
-      // Keep track of if the form has been changed.
-      $('#personalize-campaign-wizard').once('personalize-campaign-wizard-dirty', function() {
-        $(this).find(':input').change(function(e) {
-          $('#personalize-campaign-wizard').data('isDirty', true);
-        })
-      });
-      */
-
       // Handle checking for changes when using the process bar navigation
-      $('#personalize-campaign-wizard-process-bar .personalize-wizard-navigation a').once('personalize-campaign-navigation').click(function(e) {
+      $('#personalize-campaign-wizard-process-bar .personalize-wizard-navigation-item a').once('personalize-campaign-navigation').click(function(e) {
         var $parentLi = $(this).parents('li');
-        if ($parentLi.hasClass('personalize-wizard-navigation-disabled') || $parentLi.hasClass('personalize-wizard-navigation-current')) {
+        // If the clicked navigation link is available as a navigation item
+        // then set the next step, save the form, and prevent link actions.
+        if (!$parentLi.hasClass('personalize-wizard-navigation-disabled') && !$parentLi.hasClass('personalize-wizard-navigation-current')) {
+          var $campaignForm = $(this).closest('form');
+          $('input[name="override_step"]', $campaignForm).val($(this).attr('data-personalize-section'));
+          // TRICKY: Must actually click a button Drupal expects rather than
+          // a simple form.submit() or the submit function will not be called.
+          $('input#edit-save', $campaignForm).trigger('click');
```

Comment:
Is there no way to combine this approach with the one that has a concept of "isDirty", i.e. so that it only triggers the click if the form has been changed?